rfcs/2021-01-05-typography-scaling.md

```diff
@@ -0,0 +1,221 @@
+
+>**`line-height = font-size * height-factor`**
+
+And will round-off the resulting value.
+
+### **Storage of Units**
+**Global Level Tokens:** All the font attributes such as `font-family`, `line-height`, `font-size` & `font-weight` will be stored as global-level tokens.
+
+**Alias-Level Tokens:** Text colors will be stored as a alias/theme-level tokens.
+
+**Component-Level Tokens:** The main typography-scale such as `headings`, `body`, etc. will be stored as component-level tokens.
+
+### **Current Limitations**
+There are two major limitations while scaling typography to multiple breakpoints,
+1. **Need of a different typography scale.** If any product wants to bring in their own typography scale then they need to override the current typography scale (without addition/subtraction of any levels) provided by blade. This also means that you cannot add/remove levels from the base scale. This is currently being looked at by the comm. design team.
+2. **Same type scaling across all devices/breakpoints.** After the type scale is defined, we also need to define the scale across multiple device sizes. There are few approaches that we've came up with and after research from other DS/products like *GitHub*, *Eightshapes*, *Paystack*, *Stripe*, and a few more we've observed that for dashboards generally tweaking the **heading** sizes works best and the base font-size will remain same for all other devices/breakpoints.
+
+We'll look into the responsive type scale and see what approach fits best for our use case.
+
+>📝 Note: **Responsive Typography** will scale in segments. Which means if `h1` at `1440px` is `40px`, then at `991px` it will become `32px` directly rather than going from `40px -> 38px -> 36px -> 34px -> 32px...`.
+
+### **Our Approach**
+>**Multiple type-scaling across breakpoints**
+
+Our product is majorly used on mobiles/tablets/desktops. Therefore, We'll  consider only two breakpoints for the typography scale, one for mobiles & one for both desktops & tablets. Which is for devices having width `>=991px` (approx.)
+
+![Scaling: Multiple type scales ratio](./images/typography/typography-multi-type-scaling.png)
+
+In this type of scaling the modular scale changes on every breakpoint. For example: **mobiles** will use `Minor Third (1.2)` and **desktops/tablets** will use `Custom Scale (1.1)`.
+
+>📝 Note: Here we're rounding-off all the numbers for the sake of simplicity and to align the content according to the grid-system. Also, various browsers treat fractional font sizes differently.
+
+### **Example:**
+This is how it will look like across desktop & mobile.
+
+- **Type Unit Scale:** These are stored as global tokens which can be utilised by the alias-level or component-level tokens to make sense of things.
+![Type Unit Scale (Typography)](./images/typography/type-unit-scale.png)
+
+- **Typography Scale:** These are stored as component-level tokens which can be utilised by the components in the system to make meaning of things.
```

Comment:
in the image can we rename the columns as
Category -> Component Token Name
Size -> Alias

And in alias actually show `font-100` and for readability if you want you can add the actual size below `font-100`

Reply:
Also for line-height can we just show the px values? the multiplier doesn't adds any information here rather is creating more confusion.

Reply:
> in the image can we rename the columns as
> Category -> Component Token Name
> Size -> Alias
> 
> And in alias actually show font-100 and for readability if you want you can add the actual size below font-100

Didn't got this actually! Can you explain this once again? @kamleshchandnani